libraries/botframework-connector/src/auth/openIdMetadata.ts

```diff
@@ -51,11 +55,11 @@ export class OpenIdMetadata {
                 this.lastUpdated = new Date().getTime();
                 this.keys = (await getKeyResponse.json()).keys as IKey[];
             } else {
-                throw new Error(`Failed to load Keys: ${ getKeyResponse.status }`);
+                throw new AuthenticationError(`Failed to load Keys: ${ getKeyResponse.status }`, StatusCodes.BAD_REQUEST);
             }
 
         } else {
-            throw new Error(`Failed to load openID config: ${ res.status }`);
+            throw new AuthenticationError(`Failed to load openID config: ${ res.status }`, StatusCodes.NOT_FOUND);
```

Comment:
This is more that we failed to retrieve relevant information while trying to validate. Maybe 500, internal server error?